src/integrations/prefect-redis/prefect_redis/messaging.py

```diff
@@ -334,18 +340,21 @@ async def _handle_message(
         )
 
         try:
-            await handler(redis_stream_message)
-            if self.automatically_acknowledge:
-                await redis_stream_message.acknowledge()
-        except StopConsumer as e:
-            if not e.ack:
-                await self._on_message_failure(redis_stream_message, msg_id_str)
-            else:
+            try:
+                await handler(redis_stream_message)
                 if self.automatically_acknowledge:
                     await redis_stream_message.acknowledge()
-            raise
-        except Exception:
-            await self._on_message_failure(redis_stream_message, msg_id_str)
+            except StopConsumer as e:
+                if not e.ack:
+                    await self._on_message_failure(redis_stream_message, msg_id_str)
+                else:
+                    if self.automatically_acknowledge:
+                        await redis_stream_message.acknowledge()
+                raise
+            except Exception:
+                await self._on_message_failure(redis_stream_message, msg_id_str)
+        finally:
+            await self.trim_stream_if_necessary()
```

Comment:
Could you add the `finally` block to the existing `try/expect` to avoid nesting?

Reply:
eww why'd I do that?!